app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -24,13 +25,19 @@ import org.mozilla.focus.telemetry.TelemetryWrapper
 import org.mozilla.focus.telemetry.UrlTextInputLocation
 import org.mozilla.focus.utils.OnUrlEnteredListener
 import android.graphics.BitmapFactory
+import java.lang.ref.WeakReference
 
 private const val COL_COUNT = 5
 
+interface HomeDelegate {
+    fun didPressSettings()
+}
+
 /** The home fragment which displays the navigation tiles of the app. */
 class HomeFragment : Fragment() {
     lateinit var urlBar: LinearLayout
     var onUrlEnteredListener = object : OnUrlEnteredListener {} // default impl does nothing.
+    var delegate: WeakReference<HomeDelegate>? = null
```

Comment:
nit: This delegation pattern is the one you want to use, however, the Java naming is usually `Listener` and more generic, e.g. this might be a, `NavigationListener` with a method like, `onSettingsSelected` (note also: `on...` vs. `did...`). Perhaps the interface should be defined inside the outer class so that it's namespaced (because the naming is generic): HomeFragment.NavigationListener

Then again, Kotlin is new and cool and not crusty-old Java so maybe this should be a `HomeActionListener` or similar and live at the top-level. 🤔 I guess the downside is it's harder to rename with the IDE later.